docking.js

```diff
@@ -721,7 +722,9 @@ var DockedDash = GObject.registerClass({
                     this._removeBarrierTimeoutId = GLib.timeout_add(
                         GLib.PRIORITY_DEFAULT, 100, this._removeBarrier.bind(this));
                 } else {
-                    this._hide();
+                    // If an animate-out transition was delayed, check if
+                    // it is still necessary.
+                    this._hoverChanged();
```

Comment:
Replace the `_hide` call to `_hoverChanged` the same way `shouldHide` was (is) used in `_onPressureSensed` to catch any `_delayedHide` shenanigans by checking whether the delayed animate-out transition is still necessary instead of forcing the dock to hide right away and trigger a bounce from `_box`'s hover-changed signals.